### ISSUE TITLE
DatePicker: JS error with pattern containing blank and showTime="true"

### ISSUE THREAD

**Issue by ProfessionProgrammer** [primefaces/primefaces#5865]
## 1) Environment
- PrimeFaces version: 8.0
- Application server + version: WildFly 19.0.1 and jetty-9.4.27.v20200227
- Affected browsers: All

## 2) Expected behavior

It should work.

## 3) Actual behavior

Throwing Exception on JS console "Uncaught Unexpected literal at position 2" 

## 4) Steps to reproduce

click on the datepicker component.

## 5) Sample XHTML

```xml
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
	xmlns:ui="http://java.sun.com/jsf/facelets"
	xmlns:f="http://java.sun.com/jsf/core"
	xmlns:p="http://primefaces.org/ui"
	xmlns:h="http://java.sun.com/jsf/html">

<h:head>
	<title>PrimeFaces Test</title>
</h:head>
<h:body>

	<h1>#{testView.testString}</h1>
	<h:form id="frmTest">
		<p:datePicker id="startTime" value="#{testView.date}"
			hideOnDateTimeSelect="true" beforeShow="false" pattern="dd MMM yyyy"
			maxdate="#{testView.maxDate}" showTime="true" />
	</h:form>

</h:body>
</html>
```

## 6) Sample bean

```java
package org.primefaces.test;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Date;
import javax.annotation.PostConstruct;
import javax.faces.view.ViewScoped;
import javax.inject.Named;

@ViewScoped
public class TestView implements Serializable {

	private String testString;
	private Date date;
	private LocalDateTime maxDate;

	@PostConstruct
	public void init() {
		testString = "Welcome to PrimeFaces!!!";
		maxDate = LocalDateTime.now();
	}
	public String getTestString() {
		return testString;
	}
	public void setTestString(String testString) {
		this.testString = testString;
	}
	public LocalDateTime getMaxDate() {
		return maxDate;
	}
	public void setMaxDate(LocalDateTime maxDate) {
		this.maxDate = maxDate;
	}
	public Date getDate() {
		return date;
	}
	public void setDate(Date date) {
		this.date = date;
	}
}
```

**Comment by melloware** [comment]
verified on 8.0 and 9.0 with this reproducer: 
[pf-5865.zip](https://github.com/primefaces/primefaces/files/4581459/pf-5865.zip)

**Comment by christophs78** [comment]
Maybe DatePicker dislikes blanks as part of the pattern.

**Comment by melloware** [comment]
I wonder if it is related to showTime="true" even though your `pattern` does not contain time.

**Comment by ProfessionProgrammer** [comment]
But If we remove this showTime="true" attribute still it is not working, and as per the documentation pattern will work for date format only. 
https://primefaces.github.io/primefaces/8_0/#/components/datepicker

**Comment by christophs78** [comment]
generated JS:
`
pf.cw("DatePicker","widget_form_startTime",{id:"form:startTime",inline:false,userLocale:"en",dateFormat:"dd M yy",maxDate:"13 May 2020 08:19",selectionMode:"single",showOnFocus:true,appendTo:"@(body)",rangeSeparator:"\-",timeInput:false,preShow:false,showTime:true,hideOnDateTimeSelect:true});`

**Comment by christophs78** [comment]
It´s related to the blanks. `pattern="dd-MMM-yyyy"` works

https://github.com/primefaces/primefaces/blob/master/src/main/resources/META-INF/resources/primefaces/datepicker/0-datepicker.js#L722

Function "parseDate" does some "interessting" things , especially the switch in line 807 and following.

**Comment by melloware** [comment]
OK I feel like we should support spaces if Calendar does and we want people to switch from Calendar to datepicker?

Or if its not going to be supported we should throw a better error message in the console than the current one.  Maybe like a try..catch around the whole parseDate method with a console.error() saying we couldn't parse the date pattern.

Thoughts?

**Comment by christophs78** [comment]
Yep. Plan A is to support blanks.
If this fails plan B would be better error messages. Maybe already on the server during rendering.

**Comment by christophs78** [comment]
![grafik](https://user-images.githubusercontent.com/10461942/81191390-435b4c80-8fb9-11ea-83e8-c088f23a303c.png)
`parseDateTime` splits date and time at the first blank. That´s the root source of this issue.
Currently i have no good idea how to fix this. May require some refactoring.

**Comment by christophs78** [comment]
@melloware : Can you change the title of this issue? Something like "DatePicker: date-patterns containing blank and showTime="true" does not work together; throwing Exception on JS Console "Uncaught Unexpected literal at position 2"

**Comment by melloware** [comment]
Updated

**Comment by christophs78** [comment]
We have to port this commit from PrimeNG:
https://github.com/primefaces/primeng/commit/6df9fd234dc9e93436cc7449761b0f0b210bd4c4#diff-e55a2a86725724f297dc8079f0e0da5b
Exact same issue. 

Sometimes i wish PrimeTek put´s a bit more effort fixing bugs for all PrimeXXX-variations. Not only for one.

**Comment by melloware** [comment]
I almost mentioned to check PrimeNG.

**Comment by ProfessionProgrammer** [comment]
hii,
So you have fixed this issue for 9.0, so are you going to fix this for 8.0 or not?

**Comment by tandraschko** [comment]
@mertsincan can put it in the 8.0.x elite releases but 8.0 is freezed
